packages/nextjs/src/utils/clerk-js-script.tsx

```diff
@@ -0,0 +1,45 @@
+import { useClerk } from '@clerk/clerk-react';
+import { buildClerkJsScriptAttributes, clerkJsScriptUrl } from '@clerk/clerk-react/internal';
+import NextScript from 'next/script';
+import React from 'react';
+
+import { useClerkNextOptions } from '../client-boundary/NextOptionsContext';
+
+type ClerkJSScriptProps = {
+  router: 'app' | 'pages';
+};
+
+function ClerkJSScript(props: ClerkJSScriptProps) {
+  const { publishableKey, clerkJSUrl, clerkJSVersion, clerkJSVariant } = useClerkNextOptions();
+  const { domain, proxyUrl } = useClerk();
+  const options = {
+    domain,
+    proxyUrl,
+    publishableKey: publishableKey!,
+    clerkJSUrl,
+    clerkJSVersion,
+    clerkJSVariant,
+  };
+  const scriptUrl = clerkJsScriptUrl(options);
+
+  /**
+   * Notes:
+   * `next/script` in 13.x.x when used with App Router will fail to pass any of our `data-*` attributes, resulting in errors
+   * Nextjs App Router will automatically move inline scripts inside `<head/>`
+   * Using the `nextjs/script` for App Router with the `beforeInteractive` strategy will throw an error because our custom script will be mounted outside the `html` tag.
+   */
+  const Script = props.router === 'app' ? 'script' : NextScript;
+
+  return (
+    <Script
+      src={scriptUrl}
+      data-clerk-js-script
+      async
+      crossOrigin='anonymous'
+      {...buildClerkJsScriptAttributes(options)}
+      {...(props.router === 'pages' ? { strategy: 'beforeInteractive' } : {})}
```

Comment:
Do we want `beforeInteractive`? This will delay the app's hydration, based on the docs: https://nextjs.org/docs/pages/building-your-application/optimizing/scripts#strategy

I'd say we use the default setting, `afterInteractive`. We don't want to optimize clerk's loading at the expense of the host app's performance.

Reply:
I've tested both, `beforeInteractive` does not seem to have such effect. On the other hand if we use `afterInteractive` fetching of clerk-js will start after hydration making the changes of this PR to not have any effect.


![image](https://github.com/clerk/javascript/assets/19269911/97ea9943-4219-4361-886d-41b176567414)

Reply:
I thought what @BRKalow would happen as well, but after testing it seems that `beforeInteractive` does not delay hydration. This is a little surprising to me, I wonder if there's another test we could run to validate  it

Reply:
Makes sense! Nice job with the thorough testing 😃